src/hooks/useCurrentExecutionStatus.ts

```diff
@@ -0,0 +1,142 @@
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { Codec } from '@polkadot/types/types';
+import { useEffect, useState } from 'react';
+
+import { TransactionActionCreators } from '../actions/transactionActions';
+import { SOURCE, TARGET } from '../constants';
+import { useApiTargetPromiseContext } from '../contexts/ApiPromiseTargetContext';
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+import { useTransactionContext, useUpdateTransactionContext } from '../contexts/TransactionContext';
+import useDashboard from '../hooks/useDashboard';
+import useLaneId from '../hooks/useLaneId';
+import useLoadingApi from '../hooks/useLoadingApi';
+import { TransactionStatusEnum } from '../types/transactionTypes';
+import getSubstrateDynamicNames from '../util/getSubstrateDynamicNames';
+
+interface Step {
+  chainType: string;
+  label: string;
+  status: string;
+}
+
+export default function useCurrentExecutionStatus() {
+  const [nonceOfTargetFinalizedBlock, setNonceOfTargetFinalizedBlock] = useState<null | number>(null);
+  const [latestReceivedNonceRuntimeApi, setLatestReceivedNonceRuntimeApi] = useState(0);
+  const [steps, setSteps] = useState<Array<Step>>([]);
+  const { dispatchTransaction } = useUpdateTransactionContext();
+  const { api: targetApi } = useApiTargetPromiseContext();
+  const { sourceChain, targetChain } = useSourceTarget();
+  const laneId = useLaneId();
+  const areApiLoading = useLoadingApi();
+  const { currentTransaction } = useTransactionContext();
+  const {
+    bestBlockFinalized,
+    outboundLanes: { latestReceivedNonce: latestReceivedNonceOnSource }
+  } = useDashboard(SOURCE);
+  const { importedHeaders, bestBlockFinalized: bestBlockFinalizedOnTarget } = useDashboard(TARGET);
+  const { latestReceivedNonceMethodName } = getSubstrateDynamicNames(sourceChain);
+
+  useEffect(() => {
+    if (!areApiLoading || !currentTransaction) {
+      return;
+    }
+    const getNonceByHash = async () => {
+      const blockHash = await targetApi.rpc.chain.getBlockHash(bestBlockFinalizedOnTarget);
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(
+        latestReceivedNonceMethodName,
+        laneId,
+        blockHash.toJSON()
+      );
+
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      const latestReceivedNonce = latestReceivedNonceCallType.toString();
+      setNonceOfTargetFinalizedBlock(parseInt(latestReceivedNonce));
```

Comment:
Why do we go from `MessageNonce -> String -> parseInt`? Isn't there a better way to get the numeric value directly from `MessageNonce`?